test/e2e/node/pod_resize.go

```diff
@@ -453,3 +460,19 @@ var _ = SIGDescribe("Pod InPlace Resize Container", framework.WithFeatureGate(fe
 	})
 	doPodResizeAdmissionPluginsTests()
 })
+
+func waitForResourceQuota(ctx context.Context, c clientset.Interface, ns, quotaName string, used v1.ResourceList) error {
+	return wait.PollUntilContextTimeout(ctx, framework.Poll, time.Minute, false, func(ctx context.Context) (bool, error) {
+		resourceQuota, err := c.CoreV1().ResourceQuotas(ns).Get(ctx, quotaName, metav1.GetOptions{})
+		if err != nil {
+			return false, err
+		}
+		// used may not yet be calculated
+		if resourceQuota.Status.Used == nil {
+			return false, nil
+		}
+
+		framework.Logf("Resource quota status for %s has been populated", quotaName)
+		return true, nil
+	})
+}
```

Comment:
We're trying to phase out `wait.Poll*` in e2e tests. This is a better way of implementing this:
```suggestion
func waitForResourceQuota(ctx context.Context, c clientset.Interface, ns, quotaName string) error {
	return framework.Gomega().Eventually(ctx, framework.HandleRetry(func(ctx context.Context) (v1.ResourceList, error) {
		quota, err := c.CoreV1().ResourceQuotas(ns).Get(ctx, quotaName, metav1.GetOptions{})
		if err != nil {
			return nil, err
		}
		return quota.Status.Used, nil
	})).WithTimeout(framework.PollShortTimeout).ShouldNot(gomega.BeNil())
}
```
(also the `used` parameter was not used - did you mean to check it?)

Reply:
Would `gomega.BeEmpty()` be a more accurate condition to check here?

Reply:
> We're trying to phase out wait.Poll* in e2e tests. 

There is apparently much work yet to be done on that front. :stuck_out_tongue: I've updated it to use the Gomega behavior. 

> (also the used parameter was not used - did you mean to check it?)

Great catch, I stole that from the [resourceQuota tests ](https://github.com/kubernetes/kubernetes/blob/b53b9fb5573323484af9a19cf3f5bfe80760abba/test/e2e/apimachinery/resource_quota.go#L2484)and just neglected to take it out. Removed. 

> Would gomega.BeEmpty() be a more accurate condition to check here?

Yes, I believe so. I've switched it to `gomega.BeEmpty()`. I am very far from an expert on resource quotas, but looking at [where we're hitting the issue ](https://github.com/kubernetes/kubernetes/blob/a88a8238d71b160f31a6d4964c0acd5a525f85ac/staging/src/k8s.io/apiserver/pkg/admission/plugin/resourcequota/controller.go#L766) it looks like we'd fail whether we were nil or empty -- it happens when we have `resourceQuota.Status.Hard` but no matching element in `resourceQuota.Status.Used`. 

Thanks Tim!

Reply:
> There is apparently much work yet to be done on that front.

Yes, definitely. For context: https://github.com/kubernetes/kubernetes/issues/106575